go/vt/vtgate/planbuilder/route_planning.go

```diff
@@ -1019,6 +1012,39 @@ func canMergeSubqueryPlans(ctx *planningContext, a, b *route) bool {
 	return false
 }
 
+func canSelectDBAMerge(a, b *route) bool {
+	if a.eroute.Opcode != engine.SelectDBA {
+		return false
+	}
+	if b.eroute.Opcode != engine.SelectDBA {
+		return false
+	}
+
+	// safe to merge when any 1 table name or schema matches, since either the routing will match or either side would be throwing an error
+	// during run-time which we want to preserve. For example outer side has User in sys table schema and inner side has User and Main in sys table schema
+	// Inner might end up throwing an error at runtime, but if it doesn't then it is safe to merge.
+	for _, aExpr := range a.eroute.SysTableTableSchema {
+		for _, bExpr := range b.eroute.SysTableTableSchema {
+			if aExpr.String() == bExpr.String() {
+				return true
+			}
+		}
+	}
+	for _, aExpr := range a.eroute.SysTableTableName {
+		for _, bExpr := range b.eroute.SysTableTableName {
+			if aExpr.String() == bExpr.String() {
+				return true
+			}
+		}
+	}
+
+	// when we have no information from either side, they both will be routed to default keyspace, so they can be merged
+	return len(a.eroute.SysTableTableSchema) == 0 &&
+		len(a.eroute.SysTableTableName) == 0 &&
+		len(b.eroute.SysTableTableSchema) == 0 &&
+		len(b.eroute.SysTableTableName) == 0
+}
+
```

Comment:
we should also be able to merge if either side is empty. i.e. does not contain sytable table name and schema.